File: Linear Data Structure/Array/Java/ArrayTraversalWays.java
public class ArrayTraversalWays {
    public static void main(String[] args) {
        int[] arr = {5, 10, 15, 20, 25};

        // 1. Traditional for loop
        System.out.println("Using for loop:");
        for (int i = 0; i < arr.length; i++) {
            System.out.println("Element at " + i + ": " + arr[i]);
        }

        // 2. Enhanced for loop
        System.out.println("\nUsing for-each loop:");
        for (int value : arr) {
            System.out.println("Value: " + value);
        }

        // 3. While loop
        System.out.println("\nUsing while loop:");
        int i = 0;
        while (i < arr.length) {
            System.out.println("Element: " + arr[i]);
            i++;
        }
    }
}

File: Linear Data Structure/Array/C/ArrayTraversalWays.c
#include <stdio.h>

int main() {
    int arr[] = {11, 22, 33, 44, 55};
    int n = sizeof(arr) / sizeof(arr[0]);

    // Using for loop
    printf("Using for loop:\n");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }

    // Using while loop
    printf("\nUsing while loop:\n");
    int i = 0;
    while (i < n) {
        printf("%d ", arr[i]);
        i++;
    }

    // Using do-while loop
    printf("\nUsing do-while loop:\n");
    i = 0;
    do {
        printf("%d ", arr[i]);
        i++;
    } while (i < n);

    return 0;
}
